processmanager/processinfo.go

```diff
@@ -106,128 +106,65 @@ func (pm *ProcessManager) getTSDInfo(pid libpf.PID) *tpbase.TSDInfo {
-		// allocate the embedded map for this process.
-		var processName string
-		exePath, _ := os.Readlink(fmt.Sprintf("/proc/%d/exe", pid))
-		if name, err := os.ReadFile(fmt.Sprintf("/proc/%d/comm", pid)); err == nil {
-			processName = string(name)
-		}
-
-		envVarMap := make(map[string]string, len(pm.includeEnvVars))
-		if len(pm.includeEnvVars) > 0 {
-			if envVars, err := os.ReadFile(fmt.Sprintf("/proc/%d/environ", pid)); err == nil {
-				// environ has environment variables separated by a null byte (hex: 00)
-				splittedVars := strings.Split(string(envVars), "\000")
-				for _, envVar := range splittedVars {
-					keyValuePair := strings.SplitN(envVar, "=", 2)
-
-					// If the entry could not be split at a '=', ignore it
-					// (last entry of environ might be empty)
-					if len(keyValuePair) != 2 {
-						continue
-					}
-
-					if _, ok := pm.includeEnvVars[keyValuePair[0]]; ok {
-						envVarMap[keyValuePair[0]] = keyValuePair[1]
-					}
-				}
-			}
-		}
-
-		containerID, err := extractContainerID(pid)
-		if err != nil {
-			log.Debugf("Failed extracting containerID for %d: %v", pid, err)
-		}
-
-		info = &processInfo{
-			meta: ProcessMeta{
-				Name:         processName,
-				Executable:   exePath,
-				ContainerID:  containerID,
-				EnvVariables: envVarMap},
-			mappings:         make(map[libpf.Address]*Mapping),
-			mappingsByFileID: make(map[host.FileID]map[libpf.Address]*Mapping),
-			tsdInfo:          nil,
-		}
-		pm.pidToProcessInfo[pid] = info
-
-		// Insert a dummy page into the eBPF map pid_page_to_mapping_info that provides the eBPF
-		// a quick way to check if we know something about this particular process.
-		if err := pm.ebpf.UpdatePidPageMappingInfo(pid, dummyPrefix, 0, 0); err != nil {
-			return false, fmt.Errorf(
-				"failed to update pid_page_to_mapping_info dummy entry for PID %d: %v",
-				pid, err)
-		}
-		pm.pidPageToMappingInfoSize++
-	} else if mf, ok := info.mappings[m.Vaddr]; ok {
-		if *m == *mf {
-			// We try to update our information about a particular mapping we already know about.
-			return true, nil
-		}
+func (pm *ProcessManager) getPidInformation(pid libpf.PID) *processInfo {
+	if info, ok := pm.pidToProcessInfo[pid]; ok {
+		return info
 	}
 
-	info.addMapping(*m)
-
-	prefixes, err := lpm.CalculatePrefixList(uint64(m.Vaddr), uint64(m.Vaddr)+m.Length)
-	if err != nil {
-		return false, fmt.Errorf("failed to create LPM entries for PID %d: %v", pid, err)
-	}
-	numUpdates := uint64(0)
-	for _, prefix := range prefixes {
-		if err = pm.ebpf.UpdatePidPageMappingInfo(pid, prefix, uint64(m.FileID),
-			m.Bias); err != nil {
-			err = fmt.Errorf(
-				"failed to update pid_page_to_mapping_info (pid: %d, page: 0x%x/%d): %v",
-				pid, prefix.Key, prefix.Length, err)
-			break
-		}
-		numUpdates++
+	// We don't have information for this pid, so we first need to
+	// allocate the embedded map for this process.
+	var processName string
+	exePath, _ := os.Readlink(fmt.Sprintf("/proc/%d/exe", pid))
```

Comment:
Should we try to merge https://github.com/open-telemetry/opentelemetry-ebpf-profiler/pull/848 first?